crates/evm/src/lib.rs

```diff
@@ -46,60 +42,15 @@ pub mod system_calls;
 /// test helpers for mocking executor
 pub mod test_utils;
 
-/// An abstraction over EVM.
-///
-/// At this point, assumed to be implemented on wrappers around [`revm::Evm`].
-pub trait Evm {
-    /// Database type held by the EVM.
-    type DB;
-    /// Transaction environment
-    type Tx;
-    /// Error type.
-    type Error;
-
-    /// Reference to [`BlockEnv`].
-    fn block(&self) -> &BlockEnv;
-
-    /// Executes the given transaction.
-    fn transact(&mut self, tx: Self::Tx) -> Result<ResultAndState, Self::Error>;
-
-    /// Executes a system call.
-    fn transact_system_call(
-        &mut self,
-        caller: Address,
-        contract: Address,
-        data: Bytes,
-    ) -> Result<ResultAndState, Self::Error>;
-
-    /// Returns a mutable reference to the underlying database.
-    fn db_mut(&mut self) -> &mut Self::DB;
-
-    /// Executes a transaction and commits the state changes to the underlying database.
-    fn transact_commit(&mut self, tx_env: Self::Tx) -> Result<ResultAndState, Self::Error>
-    where
-        Self::DB: DatabaseCommit,
-    {
-        let result = self.transact(tx_env)?;
-        self.db_mut().commit(result.state.clone());
-
-        Ok(result)
-    }
-}
-/// Helper trait to bound [`revm::Database::Error`] with common requirements.
-pub trait Database: revm::Database<Error: core::error::Error + Send + Sync + 'static> {}
-impl<T> Database for T where T: revm::Database<Error: core::error::Error + Send + Sync + 'static> {}
+pub use alloy_evm::{Database, Evm, EvmEnv, EvmError, InvalidTxError};
```

Comment:
wondering if it's actually more helpful to not re-export all of these so libraries update their deps to point to alloy directly instead